test/approvedTransfer.js

```diff
@@ -36,30 +38,44 @@ describe("Approved Transfer", function () {
   let walletImplementation;
   let guardianManager;
   let approvedTransfer;
+  let limitModule;
   let relayerModule;
   let erc20;
   const amountToTransfer = 10000;
+  let contract;
 
   before(async () => {
     deployer = manager.newDeployer();
     const registry = await deployer.deploy(Registry);
     const guardianStorage = await deployer.deploy(GuardianStorage);
+    const limitStorage = await deployer.deploy(LimitStorage);
     guardianManager = await deployer.deploy(GuardianManager, {}, registry.contractAddress, guardianStorage.contractAddress, 24, 12);
-    approvedTransfer = await deployer.deploy(ApprovedTransfer, {}, registry.contractAddress, guardianStorage.contractAddress);
+    approvedTransfer = await deployer.deploy(ApprovedTransfer, {},
+      registry.contractAddress,
+      guardianStorage.contractAddress,
+      limitStorage.contractAddress);
     relayerModule = await deployer.deploy(RelayerModule, {},
       registry.contractAddress,
       guardianStorage.contractAddress,
-      ethers.constants.AddressZero,
+      limitStorage.contractAddress,
       ethers.constants.AddressZero);
     manager.setRelayerModule(relayerModule);
     walletImplementation = await deployer.deploy(BaseWallet);
+
+    limitModule = await deployer.deploy(TestLimitModule, {}, registry.contractAddress, guardianStorage.contractAddress, limitStorage.contractAddress);
```

Comment:
We should always try to test using the actual contracts and not sure we need this `TestLimitModule` implementation. Where it's used here it's for `getDailySpent` and `getLimit` which we can use `LimitStorage` for and also setting the Wallet's limit and daily spent, which we can get in the wallet's initialisation. Can we remove the `TestLimitModule` altogether?

Reply:
I agree on the principle but here the `TestLimitModule` is mostly introduced as a mean to initialise the limit and the dailySpent to non-zero values which is non trivial. The other alternative solution I see is adding an instance of the `TransferManager` and making a small transfer to update the dailySpent... but I find that option more complicated and equally unclean (testing the ApprovedTransfer becomes dependant of the correct behaviour of the TransferManager).

Unless I'm missing something? You mention wallet initialisation. How could we use that to set the limit and dailySpent to non-zero values?